### PR TITLE
Update to latest istio/api changes with MCP enhancements

### PR DESCRIPTION
**Pull request by ayj**
This is a non-functional change that incorporates the latest MCP API enhancements. Some fields and protos have been renamed which is a backwards incompatible change. However, Istio w/MCP has not been released yet and our integration tests upgrade Galley/Pilot/Mixer at the same time.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -295,7 +295,8 @@ ignored = [
 
 [[constraint]]
   name = "istio.io/api"
-  branch = "release-1.1"
+  branch = "release-1.1-mcp-enhancements"
```
Comment (ayj) [inline Gopkg.toml]: This change will be reverted once https://github.com/istio/api/pull/741 is merged.

**Comment by ayj** [comment]
not stale - depends on https://github.com/istio/api/pull/741

**Comment by ayj** [comment]
@ozevren @geeknoid this PR is ready for review now that istio/api#741 has been merged.

File: galley/pkg/runtime/state.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ func (s *State) apply(event resource.Event) bool {
 
 		// TODO: Check for content-wise equality
 
-		entry, ok := s.envelopeResource(event.Entry)
+		entry, ok := s.resource(event.Entry)
```
Comment (ozevren) [inline galley/pkg/runtime/state.go]: toResource or toMCPResource?

File: pkg/mcp/server/server.go [hunk 3]
```diff
@@ -87,12 +87,12 @@ type WatchResponse struct {
...
 
 	// Version of the resources in the response for the given
-	// type. The client responses with this version in subsequent
+	// type. The node responds with this version in subsequent
 	// requests as an acknowledgment.
 	Version string
 
-	// Enveloped resources to be included in the response.
-	Envelopes []*mcp.Envelope
+	// Resourced resources to be included in the response.
```
Comment (ozevren) [inline pkg/mcp/server/server.go]: Consider revising the comment.

File: galley/pkg/runtime/state.go (hunk 2 above)
Comment (ayj) [reply galley/pkg/runtime/state.go]: done

File: pkg/mcp/server/server.go (hunk 3 above)
Comment (ayj) [reply pkg/mcp/server/server.go]: fixed

File: vendor/istio.io/api/mcp/v1alpha1/mcp.proto [hunk 4]
```diff
@@ -50,8 +49,8 @@ message MeshConfigRequest {
   // below) has an independent version associated with it.
   string version_info = 1;
 
-  // The client making the request.
-  Client client = 2;
+  // The sink node making the request.
+  SinkNode sink_node = 2;
```
Comment (andraxylia) [inline vendor/istio.io/api/mcp/v1alpha1/mcp.proto]: Nit: SinkNode sounds a bit pretentious, couldn't it be simply node?

File: vendor/istio.io/api/mcp/v1alpha1/mcp.proto [hunk 5]
```diff
@@ -184,3 +183,106 @@ service AggregatedMeshConfigService {
...
+  string response_nonce = 4;
+
+  // This is populated when the previously received resources could not be applied
+  // The *message* field in *error_details* provides the source internal error
+  // related to the failure.
+  google.rpc.Status error_detail = 5;
+}
+
+// Resources do not need to include a full snapshot of the tracked
+// resources. Instead they are a diff to the state of a MCP client.
```
Comment (andraxylia) [inline vendor/istio.io/api/mcp/v1alpha1/mcp.proto]: Consider changing client to node as per the rest of changes.

File: vendor/istio.io/api/mcp/v1alpha1/mcp.proto [hunk 6]
```diff
@@ -184,3 +183,106 @@ service AggregatedMeshConfigService {
...
+// Per resource versions allow sources and sinks to track state at
+// the resource granularity. An MCP incremental session is always
+// in the context of a gRPC bidirectional stream. This allows the
+// MCP source to keep track of the state of MCP sink connected to
```
Comment (andraxylia) [inline vendor/istio.io/api/mcp/v1alpha1/mcp.proto]: Is this sink = destination?

File: vendor/istio.io/api/mcp/v1alpha1/mcp.proto [hunk 7]
```diff
@@ -184,3 +183,106 @@ service AggregatedMeshConfigService {
...
+// Resources to an RequestResources ACK or NACK.
+message Resources {
+  // The version of the response data (used for debugging).
+  string system_version_info = 1;
+
+  // Type of resource collection that is being requested, e.g.
+  //
+  // istio/networking/v1alpha3/VirtualService
+  // k8s/<apiVersion>/<kind>
+  string collection = 2;
```
Comment (andraxylia) [inline vendor/istio.io/api/mcp/v1alpha1/mcp.proto]: If this is a type of resource, why call it collection and not type?

File: vendor/istio.io/api/mesh/v1alpha1/config.proto [hunk 8]
```diff
@@ -202,46 +210,13 @@ message MeshConfig {
   // and pass to sds server, which will be used to request key/cert eventually.
   // If both enable_sds_token_mount and sds_use_k8s_sa_jwt are set to true, enable_sds_token_mount(trustworthy jwt) takes precedence.
   // This isn't supported for non-k8s case.
-  bool sds_use_k8s_sa_jwt = 30;
+  bool sds_use_k8s_sa_jwt = 29;
```
Comment (andraxylia) [inline vendor/istio.io/api/mesh/v1alpha1/config.proto]: This looks like backward incompatible change cc @quanjielin

File: vendor/istio.io/api/mesh/v1alpha1/config.proto [hunk 9]
```diff
@@ -202,46 +210,13 @@ message MeshConfig {
...
 
   // The trust domain corresponds to the trust root of a system.
   // Refer to https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain
   // Fallback to old identity format(without trust domain) if not set.
   string trust_domain = 26;
 
-  // Default service dependency settings for each workload in the
```
Comment (andraxylia) [inline vendor/istio.io/api/mesh/v1alpha1/config.proto]: Why is this being removed?

File: vendor/istio.io/api/mesh/v1alpha1/config.proto (hunk 8 above)
Comment (andraxylia) [reply vendor/istio.io/api/mesh/v1alpha1/config.proto]: I see a bunch of changes in vendor that look strange, shouldn't we update vendor separately?

File: vendor/istio.io/api/mesh/v1alpha1/config.proto (hunk 9 above)
Comment (ayj) [reply vendor/istio.io/api/mesh/v1alpha1/config.proto]: cc @rshriram - this was removed in https://github.com/istio/api/pull/742.

File: vendor/istio.io/api/mesh/v1alpha1/config.proto (hunk 8 above)
Comment (ayj) [reply vendor/istio.io/api/mesh/v1alpha1/config.proto]: cc @rshriram - this was removed in https://github.com/istio/api/pull/742.

**Comment by ayj** [comment]
https://circleci.com/gh/istio/istio/286627
> --- FAIL: TestPollingWatcher_PollingUpdates_Update (20.00s)
>     pollingWatcher_test.go:173: timeout while waiting (last error: poll error not found)

cc @ozevren

**Comment by ayj** [comment]
https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/10628/e2e-simpleTests/15419/build-log.txt
```
2019-01-10T01:11:26.799499Z	info	Command output: 
NAME                                      READY     STATUS      RESTARTS   AGE
istio-citadel-85f6b5bb58-fvzdc            1/1       Running     0          2m
istio-egressgateway-55f76cbbd8-jtpjc      1/1       Running     0          2m
istio-galley-676b87b7cd-h68t7             1/1       Running     0          2m
istio-ingressgateway-749fcb594f-ntp6p     1/1       Running     0          2m
istio-pilot-6c487d774d-hjtzq              2/2       Running     0          2m
istio-policy-7b6bdb95bb-ntxr6             2/2       Running     4          2m
istio-prometheus-test                     0/1       Error       0          2m
istio-security-test                       0/1       Completed   0          2m
istio-sidecar-injector-77d9975cf8-khxsk   1/1       Running     0          2m
istio-telemetry-d6fdbc76-nwvnp            2/2       Running     4          2m
prometheus-7549ccd956-8c5t5               1/1       Running     0          2m
```

/test e2e-simpleTests

File: galley/pkg/runtime/state.go [hunk 10]
```diff
@@ -43,7 +43,7 @@ type State struct {
 	entries     map[resource.TypeURL]*resourceTypeState
 }
 
-// per-resource-type State.
+// per-toResource-type State.
```
Comment (ozevren) [inline galley/pkg/runtime/state.go]: find-replace error. :)

**Comment by ozevren** [comment]
> https://circleci.com/gh/istio/istio/286627
> 
> > --- FAIL: TestPollingWatcher_PollingUpdates_Update (20.00s)
> > pollingWatcher_test.go:173: timeout while waiting (last error: poll error not found)
> 
> cc @ozevren

Seems like a flake. Could you open a 1.1 bug and assign it to me?

**Comment by ayj** [comment]
> Seems like a flake. Could you open a 1.1 bug and assign it to me?

https://github.com/istio/istio/issues/10851

File: vendor/istio.io/api/mcp/v1alpha1/mcp.proto (hunk 6 above)
Comment (ayj) [reply vendor/istio.io/api/mcp/v1alpha1/mcp.proto]: Yes, sink is the destination for configuration updates.

**Pull request merged**